### ISSUE TITLE
JSON formatter "embeddings" node changed to "embedding"

### ISSUE THREAD

**Issue by BuzMack** [cucumber/cucumber-jvm#1235]
## Summary

Current JSON formatter outputs the embeddings node as "embedding". Version 1.2.5 and earlier had this node as "embeddings"

## Expected Behavior

In the JSON formatted output file, I'm expecting to see this:
~~~
"embeddings": [
    {
        "mime_type": "image/url",
        "data": "aHR0cDovL2xvY2FsaG9zdC9zdGF0aWMvbG9nby5wbmc="
    }
]
~~~

## Current Behavior

Current output is as follows:
~~~
"embedding": [
    {
        "mime_type": "image/url",
        "data": "aHR0cDovL2xvY2FsaG9zdC9zdGF0aWMvbG9nby5wbmc="
    }
]

~~~

## Possible Solution

Possibly a typo during the rewrite to 2.0.0?

## Context & Motivation

This affects reports generated by cucumber reporting project: https://github.com/damianszczepanik/cucumber-reporting

Basically any attachments made during a test are present in the resulting JSON file, however the reporting tool is expecting to find them inside a node named "embeddings", and the resulting output has no attachments at all.

**Comment by BuzMack** [comment]
I suspect the simplest solution is to change JSONFormatter.java method addEmbeddingToHookMap (lines 292-298) which currently look like this:
```java
    private void addEmbeddingToHookMap(byte[] data, String mimeType) {
        if (!currentStepOrHookMap.containsKey("embedding")) {
            currentStepOrHookMap.put("embedding", new ArrayList<Map<String, Object>>());
        }
        Map<String, Object> embedMap = createEmbeddingMap(data, mimeType);
        ((List<Map<String, Object>>)currentStepOrHookMap.get("embedding")).add(embedMap);
    }
```

Change to match this instead:
```java
    private void addEmbeddingToHookMap(byte[] data, String mimeType) {
        if (!currentStepOrHookMap.containsKey("embeddings")) {
            currentStepOrHookMap.put("embeddings", new ArrayList<Map<String, Object>>());
        }
        Map<String, Object> embedMap = createEmbeddingMap(data, mimeType);
        ((List<Map<String, Object>>)currentStepOrHookMap.get("embeddings")).add(embedMap);
    }

```

**Comment by damianszczepanik** [comment]
Looks like someone has changed it and by that change the backwards compatibility was broken. Also while it contains array I don't know why someone has changed name of this node.

**Comment by gauravkarvir** [comment]
Until this is fixed and a new release version is avaialble, I have created custom class  FixScreenShotEmbedder which replaces the json node name from embedding to embeddings, execution is triggered by maven exec plugin
  <plugin>
                <groupId>org.codehaus.mojo</groupId>
                <artifactId>exec-maven-plugin</artifactId>
                <version>${mojo-exec-maven-plugin}</version>
                <executions>
                    <execution>
                        <phase>test</phase>
                        <goals>
                            <goal>java</goal>
                        </goals>
                    </execution>
                </executions>
                <configuration>
                    <mainClass>com.gk.support.FixScreenShotEmbedder</mainClass>
                    <classpathScope>test</classpathScope>
                    <arguments>
                        <argument>${project.build.directory}/cucumber</argument>
                    </arguments>
                </configuration>
            </plugin>

**Comment by mlvandijk** [comment]
@gauravkarvir You can also use the latest SNAPSHOT version, as described here: https://cucumber.io/docs/reference/jvm#installation

**Comment by gauravkarvir** [comment]
Yeah did try that, Unfortunately Snapshot releases are blocked by IT team. Hence will wait for the new release version to be available, thanks anyways...

**Comment by mlvandijk** [comment]
Ah, too bad...

**Comment by Misterhex** [comment]
looks like current snapshot version is still outputting as "embedding" instead of "embeddings".

**Comment by mpkorstanje** [comment]
[This unit test says otherwise](https://github.com/cucumber/cucumber-jvm/blame/0145029558ae5bf298c4b736916c5c7660079852/core/src/test/java/cucumber/runtime/formatter/JSONFormatterTest.java#L677)

**Comment by mlvandijk** [comment]
@Misterhex Not sure which version you are using, but the [code](https://github.com/cucumber/cucumber-jvm/blob/master/core/src/main/java/cucumber/runtime/formatter/JSONFormatter.java) says "embeddings".

**Comment by brasmusson** [comment]
> looks like current snapshot version

The current snapshot is version 2.3.2-SNAPSHOT, and it is outputting "embeddings".
